### PR TITLE
Fix attribute transition

### PR DESCRIPTION
**Pull request by Pessimistress**
For #7457 

Attribute transition is working again (#8582 highway example, scenegraph example)

#### Change List
- Update GPU transition logic to v9 buffer layout / render pass pattern
- Light refactor for better code readability
- Add types and documentation in comments to core/transitions
- Update test/apps/attribute-transition for local debugging
- Enable render tests

### CODE REVIEW

File: modules/core/src/transitions/gpu-interpolation-transition.ts [hunk 1]
```diff
@@ -38,120 +27,82 @@ export default class GPUInterpolationTransition implements GPUTransition {
...
+    let t = time / duration;
+    if (easing) {
+      t = easing(t);
     }
-    return updated;
+    const {model} = this.transform;
+    model.setUniforms({time: t});
+    // @ts-ignore
+    const gl = model.device.gl as WebGL2RenderingContext;
+    // TODO - remove after https://github.com/visgl/luma.gl/pull/2023
```
Comment (Pessimistress) [inline modules/core/src/transitions/gpu-interpolation-transition.ts:94]: Waiting on a patch with https://github.com/visgl/luma.gl/pull/2023

**Review by felixpalmer — COMMENTED** [review]
I tested the locally (overriding `countRange = [5, 5]`) and I get some strange artifacts:


https://github.com/visgl/deck.gl/assets/453755/d4080f29-7414-42b2-8ce7-6fa455f8947b

**Comment by Pessimistress** [comment]
@felixpalmer I can't reproduce it. Maybe you have outdated cache of luma locally? Try `yarn start-local --force`

File: modules/core/src/transitions/gpu-interpolation-transition.ts [hunk 2]
```diff
@@ -38,120 +27,82 @@ export default class GPUInterpolationTransition implements GPUTransition {
...
+    gl.bindBuffer(gl.ARRAY_BUFFER, null);
+
+    this.transform.run({
+      clearColor: false,
+      clearDepth: false
+    });
```
Comment (donmccurdy) [inline modules/core/src/transitions/gpu-interpolation-transition.ts:100]: Resolves the flickering:

```suggestion
    this.transform.run({discard: true});
```

Comment (donmccurdy) [reply modules/core/src/transitions/gpu-interpolation-transition.ts]: Pushed.

**Review by donmccurdy — APPROVED** [review]
After a 'spring' transition completes, it continues to shift very slightly, indefinitely, perhaps we aren't detecting when a spring transition completes? 

https://github.com/visgl/deck.gl/assets/1848368/e3725f55-5f4f-4185-b41e-06b76ef1e625

This is such a huge improvement though – I don't think that should block anything. 

The refactoring here is great, thank you! 🚀

**Pull request merged**